app/models/conversion_host.rb

```diff
@@ -3,11 +3,18 @@ class ConversionHost < ApplicationRecord
 
   acts_as_miq_taggable
 
+  VALID_RESOURCE_TYPES = %w(Vm Host).freeze
+
   belongs_to :resource, :polymorphic => true
   has_many :service_template_transformation_plan_tasks, :dependent => :nullify
   has_many :active_tasks, -> { where(:state => 'active') }, :class_name => ServiceTemplateTransformationPlanTask, :inverse_of => :conversion_host
   delegate :ext_management_system, :hostname, :ems_ref, :to => :resource, :allow_nil => true
 
+  validates :name, :presence => true
+  validates :resource_type, :presence => true, :inclusion => { :in => VALID_RESOURCE_TYPES }
+  validates :resource_id, :presence => true
+  validates :address, :presence => true, :uniqueness => true, :format => { :with => Resolv::AddressRegex }
```

Comment:
Not quite sure why we need this `address` to begin with. Is this a pick-one from the `resource.ipaddresses` list? @fdupont-redhat do you know?

Reply:
@jameswnl The `ConversionHost` model has a custom `ipaddress` method that will look at `address` first, and then use the first `resource.address` that matches the specified family, which is ipv4 by default.

Reply:
The `ipaddress` allows setting a different address than the resource, e.g. to force a specific interface. That's mainly because we can't rely on the resource IP address to be the one to connect to. In my mind, it should be unique, as I don't see the point of creating 2 conversion hosts with the same IP.

Reply:
So this is for case when the inventory refresh couldn't pick up the valid IP of the resource VM/host?

Reply:
More for the case where the resource has more than one IP address on more than one NIC. This will allow to force one of them, as `ipaddresses` returns a list and we don't have a consistent way of identifying the one to use: first ? last ? IPv4 ? IPv6 ?

Reply:
Ok.
In that case, do we want to validate that it has to be on the address list of the resource? Or we just let it be and client/caller can check if it cares

Reply:
I think it's a good check. There's no point to set it to an I P address that is not collected by refresh.

Reply:
@jameswnl @fdupont-redhat Ok, I've restricted it so that it must be one of the IP addresses of the resource.